src/Maestro/maestro-angular/src/app/page/channel/channel.component.ts

```diff
@@ -0,0 +1,51 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute } from "@angular/router";
+import { map, switchMap, tap } from 'rxjs/operators';
+
+import { FlowGraph, Channel} from 'src/maestro-client/models';
+import { Observable, of, timer, OperatorFunction } from 'rxjs';
+import { statefulSwitchMap, StatefulResult, statefulPipe } from 'src/stateful';
+import { ChannelService} from 'src/app/services/channel.service';
+import { MaestroService } from 'src/maestro-client/maestro';
+
+@Component({
+  selector: 'mc-channel',
+  templateUrl: './channel.component.html',
+  styleUrls: ['./channel.component.scss']
+})
+export class ChannelComponent implements OnInit {
+
+  constructor(private route: ActivatedRoute, private channelService: ChannelService, private maestroService: MaestroService) { }
+
+  public graph$!: Observable<StatefulResult<FlowGraph>>;
+
+  ngOnInit() {
+    const params$ = this.route.paramMap.pipe(
+      map(params => {
+        const channelId = params.get("channelId");
+
+        if ( channelId == null ) {
+          throw new Error("channelId was null");
+        } 
+
+        return {channelId};
+      }),
```

Comment:
Why make this be a new type with a single value... but the only thing we do with it is unwrap it a few lines later.  Can't we just return +channelId here, and then we don't need so many observables.

Reply:
done.